src/components/ItemAccordion/ItemAccordion.js

```diff
@@ -18,16 +19,29 @@ import {
   useData,
   useOptions,
 } from '../../hooks';
+import {
+  BARCODE_SETTING,
+  NUMBER_GENERATOR_OPTIONS_ON_EDITABLE,
+  NUMBER_GENERATOR_OPTIONS_ON_NOT_EDITABLE,
+} from '../../consts';
 
 import { validateBarcode } from '../../util';
 
-const ItemAccordion = () => {
-  const { materialTypes, loanTypes } = useData();
+const ItemAccordion = ({ change }) => {
+  const {
+    materialTypes,
+    loanTypes,
+    numberGeneratorSettings,
+  } = useData();
+
   const materialTypeOptions = useOptions(materialTypes, 'id', 'name');
   const permanentLoanTypeOptions = useOptions(loanTypes, 'id', 'name');
   const { formatMessage } = useIntl();
   const { okapi } = useStripes();
 
+  const isBarcodeDisabled = numberGeneratorSettings?.[BARCODE_SETTING] === NUMBER_GENERATOR_OPTIONS_ON_NOT_EDITABLE;
+  const showNumberGeneratorForBarcode = isBarcodeDisabled || numberGeneratorSettings?.[BARCODE_SETTING] === NUMBER_GENERATOR_OPTIONS_ON_EDITABLE;
```

Comment:
a brief comment here would help explain the circumstances when the barcode-field should be enabled, and when the generator-field should be displayed. The collection of conditions is hard to parse and feels wonky (wait, show the generator when the barcode field is disabled?!? That sort of makes sense because you'll use the generator instead of manually entering a barcode, but will a disabled field even be included in a form submission?)

Reply:
Disable manual editing of the field can be configured in the settings:
`Settings - Inventory - Number generator options - Barcode`

Reply:
My point is, should you use `disabled` or should you use `readOnly`? [A disabled field won't be part of a form submission](https://codesandbox.io/p/sandbox/beautiful-chaplygin-7ngw4h).

Reply:
Ugh, the browser and RFF are not doing us any favors here IMO, but my opinion clearly doesn't matter in the context of how a browser implements read-only 😆 . In particular I view the RFF "send all values, even for disabled fields", as a loophole I am disinclined to exploit. It just feels like the kind of thing that is going to bite us later. 

That said, I understand that `disabled` gives you exactly the UI you are looking for, and I can't argue against that. In the end, it's your call.

Reply:
Yes, I really prefer the styling of `disabled`, it is much more intuitive. Alternatively, you could rethink the styling of `readOnly` for Fields.